shared/views/RoomDirectoryView.js

```diff
@@ -17,13 +20,39 @@ class RoomDirectoryView extends TemplateView {
       vm.setSearchTerm(searchInputValueBeforeRendering);
     }
 
+    function toggleSafeSearchLabel() {
+      var text = document.getElementById('checkBox_label');
+      if (text.innerHTML === 'Safe Search is on') {
+        text.innerHTML = 'Safe Search is off';
+      } else {
+        text.innerHTML = 'Safe Search is on';
+      }
+    }
+
     const roomList = new ListView(
       {
         className: 'RoomDirectoryView_roomList',
         list: vm.rooms,
         parentProvidesUpdates: false,
       },
       (room) => {
+        let name = room.name;
+        let topic = room.topic;
+        const contains = NSFW_KEYWORDS.some((element) => {
+          if (
+            name
+              .toLowerCase()
+              .includes(
+                element.toLowerCase() || topic.toLowerCase().includes(element.toLowerCase())
+              )
+          ) {
+            return true;
+          }
+          return false;
+        });
```

Comment:
This currently matches words like `sextant` where the substring is in there but it's not what we're after.

I think we probably want to go with something like this which uses the `\b` regex word-boundary character

```suggestion
        const isRoomNsfw = NSFW_KEYWORDS.some((nsfwKeyword) => {
          const nsfwRegex = new RegExp(`/\\b${nsfwKeyword}\\b/i`);
          return room.name?.match(nsfwRegex) || room.topic?.match(nsfwRegex);
        });
```

Reply:
thank you for this suggestion. It has taught me about \b regex word-boundary character